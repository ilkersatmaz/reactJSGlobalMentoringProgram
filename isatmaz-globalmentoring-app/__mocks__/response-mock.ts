export const movieResponse = {
	data: {
		id: 221122,
		title: 'first movie of the year',
		overview:
			'overview of the video for the m ',		
		genres: ['Action', 'Horror'],
		budget: 1000,
		poster_path: 'https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg',
		release_date: '2023-01-01',
		revenue: 136907000,
		runtime: 134,
		tagline: "new and best",
		
		vote_average: 9.9,
		vote_count: 1145,
	},

	totalAmount: 1,
};
