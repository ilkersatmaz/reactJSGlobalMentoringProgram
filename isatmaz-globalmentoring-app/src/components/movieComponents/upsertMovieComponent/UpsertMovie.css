.upsert-form{
    padding: 60px;
    background-color: #232323;
}
.upsert-form-header{
    font-size: 40px;
    color:#FFFFFF;
    margin: 0;
    margin-bottom: 40px;
}
.upsert-form-div{
    display: grid;
    grid-template-columns: auto auto;
    column-gap: 50px;
    row-gap: 30px;
}

.upsert-form-item-title{
    font-size: 16px;
    line-height: 20px;
    color: #f65261;
    margin-bottom: 10px;
}

.upsert-form-item-overview{
    font-size: 16px;
    line-height: 20px;
    color: #f65261;
    margin-bottom: 10px;
    grid-column: 1 / 3;
}

.upsert-form-item-input{
    width: 95%;
    color: #FFFFFF;
    background-color: #323232;
    height: 50px;
    padding-left: 15px;
    margin-top: 10px;
}

.upsert-form-item-datepicker{
    width: 95%;
    color: #FFFFFF;
    background-color: #323232;
    padding-left: 15px;
    margin-top: 10px;
    height: 50px;
}

.upsert-form-item-multi-select{
    width: 100%;
    color: #FFFFFF;
    background-color: #323232;
    height: 50px;
    margin-top: 10px;
}

.upsert-form-item-multi-select .dropdown-container{
    background: transparent;
    border: none;
}
.upsert-form-item-multi-select .dropdown-container .dropdown-heading{
    height: 50px;
}
.upsert-form-item-multi-select .dropdown-container .panel-content{
    background-color: #323232; 
}
.upsert-form-item-multi-select .dropdown-container .panel-content li .selected{
    background-color: #323232; 
}
.upsert-form-item-multi-select .dropdown-container .panel-content li input:checked {
    accent-color: #f65261; 
}
.upsert-form-item-multi-select .dropdown-container .panel-content li:hover{
    background-color: #323232; 
}
.upsert-form-item-textarea{
    width: 96%;
    color: #FFFFFF;
    background-color: #323232;
    height: 190px;
    padding:15px 15px;
    margin-top: 10px;
}

.upsert-from-button-div{
    grid-column: 1 / 3;
    display: flex;
    justify-content: flex-end;
}

.reset-button{
    padding: 10px 20px;
    border: 1px solid #f65261;
    background-color: transparent;
    color: #f65261;
    margin-right: 15px;
}

.submit-button{
    padding: 10px 20px;
    border: 1px solid transparent;
    background-color: #f65261;
    color: #FFFFFF;
}

.confirmation-popup-content{
    width:50% !important;
    left: 0 !important;
    right: 0 !important;
    padding: 0;
    border: none !important;
    top: 100px !important;
}

.upsert-form-close{
    width: 25px;
    height: 25px;
    position: absolute;
    top: 0;
    right: 0;
    margin: 10px;
    color: #FFFFFF;
}