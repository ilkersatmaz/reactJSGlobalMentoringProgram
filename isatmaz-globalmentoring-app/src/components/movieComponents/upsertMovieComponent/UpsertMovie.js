import React, { useEffect, useState } from 'react';
import Popup from 'reactjs-popup';
import DatePicker from "react-datepicker";
import { MultiSelect } from "react-multi-select-component";
import "react-datepicker/dist/react-datepicker.css";
import ConfirmationPopup from '../confirmationComponent/ConfirmationPopup';
import {AiOutlineClose} from 'react-icons/ai';
import './UpsertMovie.css';

const genreOptions = [
  { label: "Action", value: "Action" },
  { label: "Comedy", value: "Comedy" },
  { label: "Horor", value: "Horor"},
  { label: "Drama", value: "Drama"},
  { label: "Adventure", value: "Adventure"},
  
];

const initialMovieData = {
  "title":"",
  "description":"",
  "imageUrl":"",
  "releaseDate": "",
  "url":"",
  "rating":"",
  "genre": [],
  "runTime": ""
}

function UpsertMovie (props) {
  const {setIsUpsertOpen} = props;
  const [movieData, setMovieData] = useState(props.movieData ? props.movieData : initialMovieData);
  const [selectedGenre, setSelectedGenre] = useState(props.movieData ? mapSelectedGenre(props.movieData.genre) : []);
  const [resetTriggered, setResetTriggered] = useState(false);
  useEffect(()=>{
    if(resetTriggered){
      setMovieData(props.movieData ? props.movieData : initialMovieData);
      setResetTriggered(false);
    }
  }, [resetTriggered])

  function mapSelectedGenre(selectedList){
    let multiSelectList = [];
    selectedList.map((genre)=>{
      multiSelectList.push({label : genre, value: genre});
    })
    return multiSelectList;
  }

  function getDate(releaseDate){
    if(releaseDate){
        const date=new Date(releaseDate)
        return date;
    }
    return "";
  }

 function handleChange(event) {
    setMovieData({                   
      ...movieData,    
      [event.target.name] : event.target.value       
    })
  }

 function handleSubmit(event) { 
    event.preventDefault();
  } 
  return (
    <form className='upsert-form' onSubmit={handleSubmit}>        
      <p className='upsert-form-header'> {movieData ? "EDIT" : "ADD"} YOUR MOVIE</p>
      <div className='upsert-form-div'>
        <AiOutlineClose className='upsert-form-close' onClick={()=>setIsUpsertOpen(false)}/>
        <label className='upsert-form-item-title'>
          TITLE
          <br />
          <input name="title" placeholder='title' className='upsert-form-item-input' value={movieData?.title} 
          onChange={handleChange}/>
        </label>
        <label className='upsert-form-item-title'>
          RELEASE DATE
          <br />
          <DatePicker placeholder="Select Date" className='upsert-form-item-datepicker' 
          selected={getDate(movieData?.releaseDate)} 
            onChange={(date) => {
              setMovieData({                   
                ...movieData,    
                releaseDate: date       
            })
            }}/>
        </label>
        <label className='upsert-form-item-title'>
          MOVIE URL
          <br />
          <input name="url" placeholder='https://' className='upsert-form-item-input' value={movieData?.url} 
          onChange={handleChange}/>
        </label>
        <label className='upsert-form-item-title'>
          RATING
          <br />
          <input name='rating' placeholder='7.8' className='upsert-form-item-input' value={movieData?.rating} 
          onChange={handleChange}/>
        </label>
        <label className='upsert-form-item-title'>
          GENRE
          <br />
          <MultiSelect
            className='upsert-form-item-multi-select' 
            options={genreOptions}
            value={selectedGenre}              
            label="Select Genre"
            disableSearch={true}
            hasSelectAll={false}
            onChange={(event)=>{setSelectedGenre(event)}}
          />
        </label>
        <label className='upsert-form-item-title'>
          RUNTIME
          <br />
          <input name="runTime" placeholder='minutes' className='upsert-form-item-input' value={movieData?.runTime}  
          onChange={handleChange}/>
        </label>
        <label className='upsert-form-item-overview'>
          OVERVIEW
          <br />
          <textarea name="description" placeholder='Movie Description' className='upsert-form-item-textarea' value={movieData?.description}  
          onChange={handleChange}/>
        </label>
        <div className='upsert-from-button-div'>
            <button className='reset-button' onClick={()=>{setResetTriggered(true)}}>Reset</button>
            <Popup className="confirmation-popup" trigger={<button className='submit-button'>Submit</button>} position="bottom left" arrow={false}>
              <ConfirmationPopup />
            </Popup>             
        </div>
        
      </div>        
      
    </form>
  );
}

export default UpsertMovie