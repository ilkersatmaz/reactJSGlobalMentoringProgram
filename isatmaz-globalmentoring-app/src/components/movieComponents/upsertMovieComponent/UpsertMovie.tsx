import React, { useEffect, useState } from 'react';
import Popup from 'reactjs-popup';
import DatePicker from "react-datepicker";
import { MultiSelect } from "react-multi-select-component";
import "react-datepicker/dist/react-datepicker.css";
import ConfirmationPopup from '../confirmationComponent/ConfirmationPopup';
import {AiOutlineClose} from 'react-icons/ai';
import './UpsertMovie.css';

const genreOptions = [
  { label: "Action", value: "Action" },
  { label: "Comedy", value: "Comedy" },
  { label: "Horor", value: "Horor"},
  { label: "Drama", value: "Drama"},
  { label: "Adventure", value: "Adventure"},

];

const initialMovieData = {
  "title": "",
  "tagline": "",
  "vote_average": 0,
  "vote_count": 0,
  "release_date": "",
  "poster_path": "",
  "overview": "",
  "budget": 0,
  "revenue": 0,
  "runtime": "",
  "genres": []
}

function UpsertMovie (props : any) {
  const {setIsUpsertOpen} = props;
  const [movieData, setMovieData] = useState(props.movieData ? props.movieData : initialMovieData);
  const [selectedGenre, setSelectedGenre] = useState(props.movieData ? mapSelectedGenre(props.movieData.genres) : []);
  const [resetTriggered, setResetTriggered] = useState(false);
  useEffect(()=>{
    if(resetTriggered){
      setMovieData(props.movieData ? props.movieData : initialMovieData);
      setResetTriggered(false);
    }
  }, [resetTriggered])

  function mapSelectedGenre(selectedList: any){
    let multiSelectList = [] as any;
    selectedList.map((genre : string)=>{
      multiSelectList.push({label : genre, value: genre});
    })
    return multiSelectList;
  }

  function getDate(releaseDate:  string){
    if(releaseDate){
        const date=new Date(releaseDate)
        return date;
    }
    return null;
  }

 function handleChange(event : any) {
    setMovieData({                   
      ...movieData,    
      [event.target.name] : event.target.value       
    })
  }

 function handleSubmit(event: any) { 
    event.preventDefault();
  } 
  return (
    <form className='upsert-form' onSubmit={handleSubmit}>        
      <p className='upsert-form-header'> {movieData ? "EDIT" : "ADD"} YOUR MOVIE</p>
      <div className='upsert-form-div'>
        <AiOutlineClose className='upsert-form-close' onClick={()=>setIsUpsertOpen(false)}/>
        <label className='upsert-form-item-title'>
          TITLE
          <br />
          <input name="title" placeholder='title' className='upsert-form-item-input' value={movieData?.title} 
          onChange={handleChange}/>
        </label>
        <label className='upsert-form-item-title'>
          RELEASE DATE
          <br />
          <DatePicker placeholderText="Select Date" className='upsert-form-item-datepicker' 
          selected={getDate(movieData?.release_date)} 
            onChange={(date: any) => {
              setMovieData({                   
                ...movieData,    
                release_date: date       
            })
            }}/>
        </label>
        <label className='upsert-form-item-title'>
          MOVIE URL
          <br />
          <input name="url" placeholder='https://' className='upsert-form-item-input' value={movieData?.poster_path} 
          onChange={handleChange}/>
        </label>
        <label className='upsert-form-item-title'>
          RATING
          <br />
          <input name='rating' placeholder='7.8' className='upsert-form-item-input' value={movieData?.vote_average} 
          onChange={handleChange}/>
        </label>
        <label className='upsert-form-item-title'>
          GENRE
          <br />
          <MultiSelect
            className='upsert-form-item-multi-select' 
            options={genreOptions}
            value={selectedGenre}              
            labelledBy="Select Genre"            
            disableSearch={true}
            hasSelectAll={false}
            onChange={(event: any)=>{setSelectedGenre(event)}}
          />
        </label>
        <label className='upsert-form-item-title'>
          RUNTIME
          <br />
          <input name="runTime" placeholder='minutes' className='upsert-form-item-input' value={movieData?.runtime}  
          onChange={handleChange}/>
        </label>
        <label className='upsert-form-item-overview'>
          OVERVIEW
          <br />
          <textarea name="description" placeholder='Movie Description' className='upsert-form-item-textarea' value={movieData?.overview}  
          onChange={handleChange}/>
        </label>
        <div className='upsert-from-button-div'>
            <button className='reset-button' onClick={()=>{setResetTriggered(true)}}>Reset</button>
            <Popup className="confirmation-popup" trigger={<button className='submit-button'>Submit</button>} position="bottom left" arrow={false}>
              <ConfirmationPopup />
            </Popup>             
        </div>

      </div>        

    </form>
  );
}

export default UpsertMovie