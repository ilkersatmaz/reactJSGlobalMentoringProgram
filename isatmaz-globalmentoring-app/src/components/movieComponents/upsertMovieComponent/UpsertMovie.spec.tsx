import { render } from '@testing-library/react';
import React from 'react';
import UpsertMovie from './UpsertMovie';
import { movieResponse } from '../../../../__mocks__/response-mock';

jest.mock('../../../hooks',() => ({
  useAppSelector: () => jest.fn(),
  useAppDispatch: () => jest.fn()
}))

describe('Upsert Popup component', () => {
  it("should match the snapshot", () => {  
    const { asFragment } = render(<UpsertMovie movieData={movieResponse.data}/>);
    
    expect(asFragment).toMatchSnapshot();
  });
})