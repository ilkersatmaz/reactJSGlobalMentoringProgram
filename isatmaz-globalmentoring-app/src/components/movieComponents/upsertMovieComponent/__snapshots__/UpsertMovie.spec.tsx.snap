// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Upsert Popup component should match the snapshot 1`] = `[Function]`;
